### PR TITLE
Replacing deprecated constants and adding new stories

### PR DESCRIPTION
**Pull request by dhruvv173**
## Explanation

- This PR partially fixes following issues
  - #18714 
  - #19526 
  - #17670 
- Replaced deprecated constants and created stories for `SnapRemoveWarning` and `SnapUIMarkdown`

1. `ui/components/app/snaps/snap-remove-warning`
2. `ui/components/app/snaps/snap-ui-markdown`

### `SnapRemoveWarning`
Screenshot 2023-06-13 at 7 44 14 PM

### `SnapUIMarkdown`
image
image

## Manual Testing Steps

1. Pull this branch 
2. Run `yarn storybook`
3. Check `SnapRemoveWarning` and `SnapUIMarkdown`

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Comment by dhruvv173** [comment]
hey @georgewrmarshall, could you please review this PR?
Thanks!

**Review by Goldmouth81 — APPROVED** [review]

**Comment by dhruvv173** [comment]
hey @georgewrmarshall, can we update the README as the minimum Node version required to build locally is `16.20.0`?
It was updated in #19520 
Sorry for mentioning it here, could've opened an issue but thought of asking you first.
image

**Review by DDDDDanica — DISMISSED** [review]

File: ui/components/app/app-components.scss [hunk 1]
```diff
@@ -32,7 +32,6 @@
 @import 'flask/experimental-area/index';
 @import 'snaps/snap-content-footer/index';
 @import 'snaps/snap-install-warning/index';
-@import 'snaps/snap-remove-warning/index';
```
Comment (georgewrmarshall) [inline ui/components/app/app-components.scss:35]: Removing unneeded styles

File: ui/components/app/snaps/snap-remove-warning/index.scss
Comment (georgewrmarshall) [inline ui/components/app/snaps/snap-remove-warning/index.scss:1]: Removing unneeded styles

**Review by georgewrmarshall — DISMISSED** [review]
Looking great! Made a few small updates. LGTM. Thanks for your contribution. We should get a snaps engineer to review also

**Review by NidhiKJha — DISMISSED** [review]
LGTM! But like George mentioned, it's good to have a review from snaps team and we can merge it

File: ui/components/app/snaps/snap-ui-markdown/snap-ui-markdown.js [hunk 2]
```diff
@@ -2,17 +2,17 @@ import React from 'react';
...
 } from '../../../../helpers/constants/design-system';
-import Typography from '../../../ui/typography/typography';
+import { Text } from '../../../component-library';
 
 const Paragraph = (props) => (
-  <Typography
+  <Text
     {...props}
-    variant={TypographyVariant.H6}
+    variant={TextVariant.bodySm}
```
Comment (FrederikBolding) [inline ui/components/app/snaps/snap-ui-markdown/snap-ui-markdown.js:13]: The font-size here is now too tiny.

cc @eriknson

File: ui/components/app/snaps/snap-remove-warning/snap-remove-warning.js [hunk 3]
```diff
@@ -1,51 +1,67 @@
...
-    );
-  };
+import {
+  BlockSize,
+  Display,
+  FlexDirection,
+  TextVariant,
+} from '../../../../helpers/constants/design-system';
 
+export default function SnapRemoveWarning({
```
Comment (FrederikBolding) [inline ui/components/app/snaps/snap-remove-warning/snap-remove-warning.js:23]: image
This component is no longer centered where it is used 🤔

**Review by FrederikBolding — DISMISSED** [review]
Blocking until solved!

File: ui/components/app/snaps/snap-ui-markdown/snap-ui-markdown.js (hunk 2 above)
Comment (eriknson) [reply ui/components/app/snaps/snap-ui-markdown/snap-ui-markdown.js:13]: Font-size should be `bodyMd` (baseline, 14px)

File: ui/components/app/snaps/snap-remove-warning/snap-remove-warning.js (hunk 3 above)
Comment (eriknson) [reply ui/components/app/snaps/snap-remove-warning/snap-remove-warning.js:23]: Shouldn't the body text font for this modal be our `bodyMd` baseline version? (@georgewrmarshall)

Comment (georgewrmarshall) [reply ui/components/app/snaps/snap-remove-warning/snap-remove-warning.js:23]: The updated modal isn't centered it prevents it from jumping around when the content of the modal changes. Also yes the text should use `bodyMd`

Comment (georgewrmarshall) [reply ui/components/app/snaps/snap-remove-warning/snap-remove-warning.js:23]: Updated text variant here https://github.com/MetaMask/metamask-extension/pull/19563/commits/20f698df12c88196235a936f332b4b0608af55f8

File: ui/components/app/snaps/snap-ui-markdown/snap-ui-markdown.js (hunk 2 above)
Comment (georgewrmarshall) [reply ui/components/app/snaps/snap-ui-markdown/snap-ui-markdown.js]: Updated https://github.com/MetaMask/metamask-extension/pull/19563/commits/20f698df12c88196235a936f332b4b0608af55f8

**Review by georgewrmarshall — APPROVED** [review]
LGTM! Thanks for your contribution @dhruvv173

**Review by garrettbear — APPROVED** [review]

**Review by Mrtenz — APPROVED** [review]

**Pull request merged**